crates/pathfinder/src/rpc/v02/method/block_hash_and_number.rs

```diff
@@ -0,0 +1,62 @@
+use anyhow::Context;
+use stark_hash::StarkHash;
+
+use crate::{rpc::v02::RpcContext, storage::StarknetBlocksTable};
+
+type BlockNumber = u64;
+
+#[derive(serde::Serialize)]
+pub struct BlockNumberResult {
+    pub block_hash: StarkHash,
+    pub block_number: BlockNumber,
+}
+
+crate::rpc::error::generate_rpc_error_subset!(BlockNumberError: BlockNotFound);
+
+pub async fn block_hash_and_number(
+    context: RpcContext,
+) -> Result<BlockNumberResult, BlockNumberError> {
+    let storage = context.storage.clone();
+    let span = tracing::Span::current();
+
+    let jh = tokio::task::spawn_blocking(move || {
+        let _g = span.enter();
+        let mut db = storage
+            .connection()
+            .context("Opening database connection")?;
+        let tx = db.transaction().context("Creating database transaction")?;
+
+        if let Some((hash, number)) = StarknetBlocksTable::get_latest_hash_and_number(&tx)
+            .context("Reading latest block hash and number from database")?
+        {
+            Ok(BlockNumberResult {
+                block_hash: hash.0,
+                block_number: number.get(),
+            })
+        } else {
+            Err(BlockNumberError::BlockNotFound)
+        }
```

Comment:
A more functional approach - imo it does make the error handling more clear.
```suggestion
        StarknetBlocksTable::get_latest_hash_and_number(&tx)
            .context("Reading latest block hash and number from database")?
            .map(|(block_hash, block_number)| BlockNumberResult { 
                block_hash,
                block_number,
            })
            .ok_or(BlockNumberError::BlockNotFound)
```

Reply:
Done